common/session/class.BasicSession.php

```diff
@@ -142,6 +142,10 @@ public function getDataLanguage()
      */
     public function getInterfaceLanguage()
     {
+        if (PHPSession::singleton()->hasAttribute('overrideInterfaceLanguage')) {
```

Comment:
There's a wrapper around PHPSession that we could rely on to avoid make static calls so we further replace it easily:

`$this->getServiceLocator()->get(SessionService::SERVICE_ID)->hasAttribute('overrideInterfaceLanguage')
`
(from oat\oatbox\session\SessionService).

Also, what do you think about adding a new method like 'setInterfaceLanguage' to this interface instead? Are there blockers? Because I think it might be more obvious than reading a session attribute

Reply:
I might be wrong, but I don't think `SessionService` provides such a method.

Reply:
Unfortunately, I can not use this service because it provides a limited session object. I could not find any other option to access PHPSession other than that. :(